executor/src/constant_evaluator/mod.rs

```diff
@@ -69,10 +70,19 @@ fn generate_values<T: FieldElement>(
             };
             let index_expr;
             let e = if let Some(index) = index {
-                index_expr = Expression::IndexAccess(IndexAccess {
-                    array: e.clone().into(),
-                    index: Box::new(Expression::Number(index.into(), None)),
-                });
+                index_expr = Expression::IndexAccess(
+                    IndexAccess {
+                        array: e.clone().into(),
+                        index: Box::new(Expression::Number(
+                            Number {
+                                value: index.into(),
+                                type_: None,
+                            },
+                            SourceRef::unknown(),
```

Comment:
I'm not sure about these two unknowns, should I take the value from e? Or does it not make sense in this context?